org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/variables/IVariableFormatter.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.adapter.variables;
+
+import com.sun.jdi.Type;
+import com.sun.jdi.Value;
+import org.eclipse.jdt.ls.debug.adapter.formatter.ITypeFormatter;
+import org.eclipse.jdt.ls.debug.adapter.formatter.IValueFormatter;
+
+import java.util.Map;
+
+public interface IVariableFormatter {
+    /**
+     * Register a type formatter. Be careful about the priority of formatters, the formatter with the largest
+     * priority which accepts the type will be used.
+     *
+     * @param typeFormatter the type formatter
+     * @param priority      the priority for this formatter
+     */
+    void registerTypeFormatter(ITypeFormatter typeFormatter, int priority);
+
+    /**
+     * Register a value formatter. Be careful about the priority of formatters, the formatter with the largest
+     * priority which accepts the type will be used.
+     *
+     * @param formatter the value formatter
+     * @param priority  the priority for this formatter
+     */
+    void registerValueFormatter(IValueFormatter formatter, int priority);
+
+    /**
+     * Get the matched value formatter for the specified type.
+     * @param type the type of the valuer
+     * @param options the options about the formatters
+     * @return the matched value formatter, if no formatter matches,
+     * <code>IllegalArgumentException</code> will be thrown
+     */
+    IValueFormatter getValueFormatter(Type type, Map<String, Object> options);
```

Comment:
I don't see the use case of this one. This is called internally by valueToString or strongToValue by some implementation. Others will not call it directly. Suggest remove.